core/src/main/scala/cilib/RVar.scala

```diff
@@ -174,12 +174,30 @@ object RVar extends RVarInstances {
           .traverse(x => Dist.uniformInt(Interval(0, x)))
 
       backsaw
-        .map(_.foldLeft(List.empty[A])((a, c) =>
-          F.index(xs, c) match {
-            case Some(i) => a :+ i
-            case None    => sys.error("Shouldn't be possible")
-        }))
-        .map(_.some)
+        .map(l => {
+          def dropIndex(target: Int, l: List[A]): List[A] = {
+            def innerDropIndex(count: Int, current: List[A]): List[A] =
+              current match {
+                case Nil => Nil
+                case x :: xs =>
+                  if (count == target) xs else x :: innerDropIndex(count + 1, xs)
+              }
+
+            innerDropIndex(0, l)
+          }
+
+          def go(indexes: List[Int], current: List[A]): List[A] =
```

Comment:
What happens if, for example we have a `List` of size `n` and we generate random indices such as `n-2` and `n-1`. And since the method `go` will remove indices already visited then the index `n-1` will become invalid. Is this correct?

Reply:
I don't follow your thinking here. The values in `backsaw` are based on a "reducing list size", so even if the same same value is randomly selected, the selected element will be within range and the elements will always be shrinking as selections take place.

Reply:
I'm with you now. That's very cool. One final thought though, because `backsaw` uses the reducing list size to generate random indices do we not end up giving preference to indices `0 to n - 1` as they would have the most opportunities to be generated?

Reply:
I understand now that since already visited elements are being removed that the elements closer to `xs.length` will also get a chance of selected fairly. Will Approve the PR